### ISSUE TITLE
Pass binary to `cast_attachments`

### ISSUE THREAD

**Issue by darkbaby123** [stavro/arc_ecto#54]
There's a referenced [issue](https://github.com/stavro/arc/issues/107) in arc but I think it's arc_ecto's responsibility so I create an issue here.

My requirement is uploading file as base64 encoded string to the server. It's common for small files and is easy to work with JSON API. After googling I found arc support binary, so I thought I can decode base64 to binary and save the file.

arc support passing binary like this `Avatar.store(%{filename: "file.png", binary: binary})` but there's no way to pass binary in arc_ecto, The `cast_attachments(%{filename: "file.png", binary: binary})` fails. I checked the source and it seems the arc_ecto only support `Plug.Upload` struct. Is there a way or workaround to achieve this?

**Comment by philipgiuliani** [comment]
I am doing the same in my Project. Version `v0.5` broke this behaviour and I am not aware of a workaround. If you use version `v0.5.0-rc1` or `v0.4.0` everything should be fine.

@stavro Is there a way to do this now?

**Comment by yogipatel** [comment]
I agree, this would be a nice feature to have. I can try to make a PR if you'd like, @stavro.

@darkbaby123 @philipgiuliani A workaround you can use for now is to write the data out to a temp file and pass that temp file's path in.

**Comment by Clyvv** [comment]
Hi, im using v0.7.0 and still could not pass  binary data to cast_attachments.  Was this ever solved?  Also I tried using Plug.Upload by creating a temp file but it seems it works at times and does not work the other times. Can someone help me out here. My function that converts my base64 to Plug.Upload is as follow
`
def process_image(image_base64) do
 {:ok, image_binary} = Base.decode64(image_base64)
filename =
      image_binary
      |> image_extension()
      |> unique_filename()

case File.write!("candidate_pics/#{filename}", image_binary, [:binary]) do
      :ok ->
        %{filename: filename, binary: image_binary}

        %Plug.Upload{
          content_type: "image/#{image_extension(image_binary)}",
          filename: filename,
          path: Path.expand("candidate_pics/#{filename}") |> Path.absname()
        }

      {:error, reason} ->
        {:error, reason}
    end
end
`

**Comment by jalcine** [comment]
Going to bump this issue and the accompanying PR #87 that fixes this.

For now, I'm just going to work off that branch until this is merged.

Nice work with the project @stavro, I know you must be busy.

**Comment by jeremyjh** [comment]
`arc` works fine if you pass a map `%{binary: <<your bytes>>, filename: "filename.txt"}`. The problem in  `arc_ecto` is `cast_attachment` [matches the file arg](https://github.com/stavro/arc_ecto/blob/8be0091c29b9ba8e6c8c95065a86b71aaf96b2e9/lib/arc_ecto/schema.ex#L40) against `%{__struct__: Plug Upload}`.

So as a workaround, you can pass:

```elixir
    %{
      __struct__: Plug.Upload,
      binary: bytes,
      filename: "PO - " <> activity.name <> ".xlsx"
    }
```

In place of an actual `Plug.Upload`. Ugly as sin but it works.
